crates/vm/levm/src/precompiles.rs

```diff
@@ -95,12 +98,70 @@ pub fn execute_precompile(current_call_frame: &mut CallFrame) -> Result<Bytes, V
     Ok(result)
 }
 
-fn ecrecover(
-    _calldata: &Bytes,
-    _gas_for_call: U256,
-    _consumed_gas: &mut U256,
+fn fill_with_zeros(slice: &[u8]) -> [u8; 128] {
+    let mut result = [0; 128];
+
+    let n = slice.len().min(128);
+
+    let trimmed_slice = slice.get(..n).unwrap_or_default();
+    result
+        .get_mut(..n)
+        .unwrap_or_default()
+        .copy_from_slice(trimmed_slice);
+
+    result
+}
+
+pub fn ecrecover(
+    calldata: &Bytes,
+    gas_for_call: U256,
+    consumed_gas: &mut U256,
 ) -> Result<Bytes, VMError> {
-    Ok(Bytes::new())
+    // Consume gas
+    *consumed_gas = consumed_gas
+        .checked_add(ECRECOVER_COST.into())
+        .ok_or(PrecompileError::GasConsumedOverflow)?;
+
+    if gas_for_call < *consumed_gas {
+        return Err(VMError::PrecompileError(PrecompileError::NotEnoughGas));
+    }
+
+    // If calldata does not reach the required length, we should fill the rest with zeros
+    let calldata = fill_with_zeros(calldata);
+
+    let hash = calldata
+        .get(0..32)
+        .ok_or(PrecompileError::ParsingInputError)?;
```

Comment:
I think these parsing input errors should halt execution, not revert. So maybe they should be considered Internal Errors.

Reply:
Done in [Use Internal::SlicingError when slicing fails](https://github.com/lambdaclass/ethrex/pull/1511/commits/95d04be51358c52d017a15fa432850a899a7961c)